### DISCUSSION TITLE
Prisma MySQL SELECT...FOR UPDATE, SELECT...FOR SHARE

### DISCUSSION THREAD

**Discussion by backend-sungkun** [prisma/prisma#25048] [Q&A] [answered]
### Question

I’m using MySQL with Prisma, and I’m wondering if SELECT ... FOR UPDATE or SELECT ... FOR SHARE statements can be made by Prisma? (I wonder this because without these statements, MySQL repeatable-read transaction level is almost same with serialize level, that is , no phantom read occurs.)

**Comment by ludralph** [comment, marked as answer]
Hi @backend-sungkun 👋 

Currently, Prisma Client does not natively support `SELECT ... FOR UPDATE` or `SELECT ... FOR SHARE` statements. This feature has been requested and is tracked in the following GitHub issue: [SELECT FOR UPDATE](https://github.com/prisma/prisma/issues/8580).

However, you can achieve similar functionality using [Raw SQL ](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access)queries with Prisma. 

Here is an example of how you might use a raw query to achieve SELECT ... FOR UPDATE:

```js
const result = await prisma.$executeRaw`SELECT * FROM your_table WHERE id = 1 FOR UPDATE`;
```

You can leave a 👍 on the issue to help our Engineers prioritize it.
